app/src/main/java/org/jetbrains/kotlinx/dl/example/app/MainActivity.kt

```diff
@@ -44,30 +37,15 @@ class MainActivity : AppCompatActivity(), OnItemSelectedListener {
         val modelsSpinnerAdapter = ArrayAdapter(
             this,
             android.R.layout.simple_spinner_item,
-            Pipelines.values().map { it.name })
+            Pipelines.values().map { it.name }
+        )
         modelsSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
 
         with(models) {
             adapter = modelsSpinnerAdapter
-            onItemSelectedListener = object : OnItemSelectedListener {
-                override fun onItemSelected(
-                    parent: AdapterView<*>?,
-                    view: View?,
-                    position: Int,
-                    id: Long
-                ) {
-                    pipelineAnalyzer?.setPipeline(position)
-                }
-
-                override fun onNothingSelected(parent: AdapterView<*>?) {
-                    pipelineAnalyzer?.clear()
-                }
-            }
-            prompt = "Select model"
-            gravity = Gravity.CENTER
-
+            onItemSelectedListener = this@MainActivity
```

Comment:
I often see this pattern in the android examples, which makes me sad. Methods `onItemSelected` are not meant to be public, they are a implementation detail and only meant to be called by this specific spinner class and not exposed to the world.